src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/LocalFolderUtility.cs

```diff
@@ -668,6 +668,68 @@ public static IEnumerable<FileInfo> GetNupkgsFromFlatFolder(string root, ILogger
             yield break;
         }
 
+        public static FeedType GetLocalFeedType(string root, ILogger log)
+        {
+            if (root == null)
+            {
+                throw new ArgumentNullException(nameof(root));
+            }
+
+            if (log == null)
+            {
+                throw new ArgumentNullException(nameof(log));
+            }
+
+            // Check for package files one level deep.
+            DirectoryInfo rootDirectoryInfo = GetAndVerifyRootDirectory(root);
+
+            try
+            {
+
+                if (rootDirectoryInfo.EnumerateFiles(NupkgFilter, SearchOption.TopDirectoryOnly).Any())
```

Comment:
Add a comment explaining the order here and how it checks v2 and v3 at the same time